integration/spark/app/src/test/java/io/openlineage/spark/agent/lifecycle/plan/CustomEnvironmentVariablesCaptureTest.java

```diff
@@ -85,6 +85,9 @@ void testCustomVariableCapture(@TempDir Path tempDir, SparkSession spark)
             .filter(
                 e ->
                     e.getEventType().equals(EventType.START)
+                        && e.getJob().getFacets() != null
+                        // Exclude SparkApplication start event because env variable was set after
+                        && e.getJob().getFacets().getJobType() != null
```

Comment:
Would it make sense to extend some of the existing integration tests to verify that new facets are present within application start event?

Reply:
Added tests for emitting `spark_properties`, `environment-properties` and `processing_engine` facets for Spark application & job events.